java/arcs/core/entity/VariableEntityBase.kt

```diff
@@ -0,0 +1,50 @@
+package arcs.core.entity
+
+import arcs.core.common.Referencable
+import arcs.core.data.FieldName
+import arcs.core.data.RawEntity
+import arcs.core.data.Schema
+
+/**
+ * A base [Entity] for type variables.
```

Comment:
Maybe a bit more specific: 

`A base [Entity] to access data from constrained type variables`

Reply:
I like it, but as a nit: sometimes the TVs won't be constrained.